xbmc/addons/RepositoryUpdater.cpp

```diff
@@ -213,12 +230,15 @@ void CRepositoryUpdater::ScheduleUpdate()
   if (!m_addonMgr.HasAddons(ADDON_REPOSITORY))
     return;
 
-  auto prev = LastUpdated();
-  auto next = std::max(CDateTime::GetCurrentDateTime(), prev + interval);
-  int delta = std::max(1, (next - CDateTime::GetCurrentDateTime()).GetSecondsTotal() * 1000);
-
-  CLog::Log(LOGDEBUG,"CRepositoryUpdater: previous update at %s, next at %s",
-      prev.GetAsLocalizedDateTime().c_str(), next.GetAsLocalizedDateTime().c_str());
+  int delta{1};
+  const auto nextCheck = ClosestNextCheck();
+  if (nextCheck.IsValid())
+  {
+    // Repos were already checked once and we know when to check next
+    delta = std::max(1, (nextCheck - CDateTime::GetCurrentDateTime()).GetSecondsTotal() * 1000);
```

Comment:
@yol In case Kodi is diconnected from network, I experience high CPU load caused by kodi spawning repo updates every 1 millisecond. This renders Kodi almost unuseless. Could you pleaes tale a look and fix this, thanks.

Reply:
Can you try inserting this into CRepositoryUpdateJob::DoWork()?

After `auto status = m_repo->FetchIfChanged(oldChecksum, newChecksum, addons, recheckAfter);` insert

`if (status == CRepository::STATUS_ERROR) { recheckAfter = 60; }`

Reply:
This will definitely fix the problem. I would go with 1 hour, not one 1 minute, though.

Reply:
How about this?

```diff
diff --git a/xbmc/addons/Repository.cpp b/xbmc/addons/Repository.cpp
index 22fd0889e8..76764c59dc 100644
--- a/xbmc/addons/Repository.cpp
+++ b/xbmc/addons/Repository.cpp
@@ -239,6 +239,7 @@ CRepository::FetchStatus CRepository::FetchIfChanged(const std::string& oldCheck
       int recheckAfterThisDir;
       if (!FetchChecksum(dir.checksum, part, recheckAfterThisDir))
       {
+        recheckAfter = 1 * 60 * 60; // retry after 1 hour
         CLog::Log(LOGERROR, "CRepository: failed read '%s'", dir.checksum.c_str());
         return STATUS_ERROR;
       }
```

Reply:
Also fine, do you want to PR?

Reply:
Yep, will do.